sycl/source/detail/graph_impl.hpp

```diff
@@ -194,66 +263,93 @@ struct graph_impl {
         "No event has been recorded for the specified graph node");
   }
 
-  /// Adds subgraph nodes from an executable graph to this graph, returns
-  /// an empty node is used to schedule dependencies on this sub graph.
+  /// Adds subgraph nodes from an executable graph to this graph
+  /// @return An empty node is used to schedule dependencies on this sub graph.
   std::shared_ptr<node_impl>
   add_subgraph_nodes(const std::list<std::shared_ptr<node_impl>> &NodeList);
+
+  /// Query for the context tied to this graph
+  /// @return context associated with graph
   sycl::context get_context() const { return MContext; }
 
+  /// List of root nodes
   std::set<std::shared_ptr<node_impl>> MRoots;
-  std::shared_ptr<graph_impl> MParent;
 
 private:
-  // Context associated with this graph.
+  /// Context associated with this graph.
   sycl::context MContext;
-  // Device associated with this graph. All graph nodes will execute on this
-  // device.
+  /// Device associated with this graph. All graph nodes will execute on this
+  /// device.
   sycl::device MDevice;
-  // Unique set of queues which are currently recording to this graph.
+  /// Unique set of queues which are currently recording to this graph.
   std::set<std::shared_ptr<sycl::detail::queue_impl>> MRecordingQueues;
-  // Map of events to their associated recorded nodes.
+  /// Map of events to their associated recorded nodes.
   std::unordered_map<std::shared_ptr<sycl::detail::event_impl>,
                      std::shared_ptr<node_impl>>
       MEventsMap;
 };
 
+/// Class representing the implementation of command_graph<executable>
 class exec_graph_impl {
 public:
+  /// Constructor
+  /// @param Context context to create graph with
+  /// @param GraphImp Modifiable graph implementation to create with
   exec_graph_impl(sycl::context Context,
                   const std::shared_ptr<graph_impl> &GraphImpl)
       : MSchedule(), MGraphImpl(GraphImpl), MPiCommandBuffers(),
         MPiSyncPoints(), MContext(Context) {}
+
+  /// Destructor
```

Comment:
Nitpick but I think this is probably unnecessary. Having it doesn't necessarily hurt anything other than create a bit of noise I guess, but perhaps better not to add that where possible.

Reply:
Good point, right now the comment doesn't communicate any extra info. Rather than deleting it, I could add to the detailed description that the destructor releases the underlying PI command-buffers, which would give the comment informative value.